pilot/helpers/Debugger.py

```diff
@@ -66,19 +68,22 @@ def debug(self, convo, command=None, user_input=None, issue_description=None, is
                         test_command=command,
                         test_after_code_changes=True,
                         continue_development=False,
-                        is_root_task=is_root_task)
+                        is_root_task=is_root_task,
+                        continue_from_step=len(completed_steps)
+                    )
 
                     if 'step_index' in result:
-                        # result['running_processes'] = running_processes
                         result['os'] = platform.system()
                         step_index = result['step_index']
-                        result['completed_steps'] = steps[:step_index]
+                        completed_steps = steps[:step_index+1]
+                        result['completed_steps'] = completed_steps
                         result['current_step'] = steps[step_index]
                         result['next_steps'] = steps[step_index + 1:]
-                        result['current_step_index'] = step_index
+                        result['current_step_index'] = step_index + len(completed_steps)
 
                         # Remove the previous debug plan and build a new one
                         convo.remove_last_x_messages(2)
+                        # todo before updating task first check if update is needed
```

Comment:
how will this be checked?

Reply:
add new prompt asking LLM to reply with "NEEDS_UPDATE" if it is needed to change next steps or to reply with "DONE" if next steps are ok. In case it answers with "NEEDS_UPDATE" then we call update_task.prompt